clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -5892,8 +5892,6 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
         getTarget().getTriple().isAMDGCN() ||
         (getTarget().getTriple().isSPIRV() &&
          getTarget().getTriple().getVendor() == Triple::VendorType::AMD)) {
-      if (getLangOpts().OpenMPIsTargetDevice)
-        return EmitOpenMPDevicePrintfCallExpr(E);
```

Comment:
Is EmitDevicePrintfCallExpr now dead and deletable?

Reply:
Not in this patch. It's still used for `NVPTX` because CUDA doesn't know it can do varargs yet. We could replace that with the same code I put in `LibC.h` here in one of the CUDA headers if we wanted to. OpenCL and HIP also still use the ROCm Device Libs for this.